File: cpp-answers/263.丑数.cpp
/*
 * @Author: your name
 * @Date: 2020-06-08 09:17:39
 * @LastEditTime: 2020-06-08 09:18:35
 * @LastEditors: your name
 * @Description: In User Settings Edit
 * @FilePath: \leetcode刷题\263.丑数.cpp
 */
/*
 * @lc app=leetcode.cn id=263 lang=cpp
 *
 * [263] 丑数
 */

// @lc code=start
class Solution
{
public:
    bool isUgly(int num)
    {
        for (int i = 2; i < 6 && num; i++)
        {
            while (num % i == 0)
                num /= i;
        }
        return num == 1;
    }
};
// @lc code=end

File: rust-answers/263.丑数.rs
/*
 * @lc app=leetcode.cn id=263 lang=rust
 *
 * [263] 丑数
 */

// @lc code=start
impl Solution {
    pub fn is_ugly(n: i32) -> bool {
        match n {
            _ if n <= 0 => false,
            mut num => {
                for p in [2, 3, 5] {
                    while num % p == 0 {
                        num /= p;
                    }
                }
                num == 1
            }
        }
    }
}
// @lc code=end

